api/tasks.go

```diff
@@ -595,11 +595,18 @@ type DispatchPayloadConfig struct {
 	File string
 }
 
+type TaskLifecycle struct {
+	Hook       string        `mapstructure:"hook"`
+	BlockUntil string        `mapstructure:"block_until"`
```

Comment:
Could we maybe include the supported values in a comment here?

Reply:
I duplicated the supported values into the api package so it doesn't depend on structs.